pkg/config/expand.go

```diff
@@ -710,11 +711,15 @@ func expandSimpleVariable(
 			if !slices.Contains(toMod.Outputs, ref.Name) {
 				toMod.Outputs = append(toMod.Outputs, ref.Name)
 			}
-			// TODO: expandedVariable = fmt.Sprintf("((var.%s_%s))", ref.Name, ref.ID)
-			return "", fmt.Errorf("%s: %s is an intergroup reference",
-				errorMessages["varInAnotherGroup"], context.varString)
+			switch fromMod.Kind {
+			case "terraform":
+				expandedVariable = fmt.Sprintf("((var.%s_%s))", ref.Name, ref.ID)
+			case "packer":
+				expandedVariable = "((TEST))"
```

Comment:
I would have expected error on this case. I am betting we error somewhere else?

Reply:
This code is a proxy for the future functionality of adding corresponding inputs. I believe this block is eventually going to modify a struct that tracks "intergroup data" to collect from previous groups to guide the user (or automated solution) on how to apply values from previous groups. This will be true for both Packer and Terraform. Because of that, it probably will not error for Packer and we will have to return some sort of valid string value that will not be written to disk in `defaults.auto.pkrvars.hcl`. I am considering the empty Toolkit literal `"(())"`.

Reply:
Ahh. This is a packer module referencing a terraform module. I was thinking the other way around. Correct? If so then it makes sense to me why we wouldn't error.

I am still not sure why you would leave it empty. Would this not be some variable name that we are feeding into Packer. In my current understanding I would expect it to look similar to the Terraform reference. Maybe I am missing something critical about how Packer does variables.

Reply:
Yes, this covers the use case of Packer using Terraform (e.g. a subnetwork created by VPC module with known firewall properties). The critical differences between Packer "templates" and Terraform modules, which we gloss over by calling each Toolkit modules:

1. Packer has no concept of state
1. Packer has no native concept of outputs
  * thus we cannot "use" Packer modules
  * there is a task to define a Toolkit notion of Packer outputs (specifically to get image_name)
1. Packer has no mechanism by which to refer to a remote template (i.e. a Packer template has no child templates)

(3) means we are stuck with the input variables present in the Packer template itself and cannot construct new ones like we do for the Terraform root modules. There is a later section of code (in modulewriter) that writes `default.auto.prkvals.hcl` which is the equivalent of `terraform.tfvars`. This code will attempt to evaluate anything with pattern `((var.variable_name))` in the "context" of our deployment variables (`vars.*`) so that it imitates what we do for `terraform.tfvars`.

We should evaluate whether the better place to do that is in this section of code in the first place. This is a tough "[traversal](https://pkg.go.dev/github.com/hashicorp/hcl2/hcl#Traversal)" through the code. I will create a task do that.

Either way, for intergroup references, we need to have something here that indicates the variable should not be written to `defaults.auto.pkrvals.hcl` at all and needs to be collected (by user or by ghpc) from prior Terraform state before running `packer build`.

Hopefully this essay clarifies the situation.

Reply:
I think I understand conceptually but am still struggling to see why this leads us to have `(())`. 

IIUC this was part of the commit that was popped off of the top, correct? If so feel free to close this comment and we can discuss further later.

Reply:
That is correct. This change is not part of the PR and will be part of the subsequent work to glue outputs and inputs together across groups.